core/services/eth/errors.go

```diff
@@ -42,146 +42,109 @@ func (s *SendError) CauseStr() string {
-	parGasLimitExceeded = regexp.MustCompile("^Transaction cost exceeds current gas limit. Limit:")
-	parInvalidSignature = regexp.MustCompile("^Invalid signature")
-	parInvalidGasLimit  = "Supplied gas is beyond limit."
-	parSenderBanned     = "Sender is banned in local queue."
-	parRecipientBanned  = "Recipient is banned in local queue."
-	parCodeBanned       = "Code is banned in local queue."
-	parNotAllowed       = "Transaction is not permitted."
-	parTooBig           = "Transaction is too big, see chain specification for the limit."
-	parInvalidRlp       = regexp.MustCompile("^Invalid RLP data:")
-)
-
-// Geth and geth-compatible errors
-var (
-	gethNonceTooLow                       = regexp.MustCompile(`(: |^)nonce too low$`)
-	gethReplacementTransactionUnderpriced = regexp.MustCompile(`(: |^)replacement transaction underpriced$`)
-	gethKnownTransaction                  = regexp.MustCompile(`(: |^)(?i)(known transaction|already known)`)
-	gethTransactionUnderpriced            = regexp.MustCompile(`(: |^)transaction underpriced$`)
-	gethInsufficientEth                   = regexp.MustCompile(`(: |^)(insufficient funds for transfer|insufficient funds for gas \* price \+ value|insufficient balance for transfer)$`)
-	gethTxFeeExceedsCap                   = regexp.MustCompile(`(: |^)tx fee \([0-9\.]+ ether\) exceeds the configured cap \([0-9\.]+ ether\)$`)
-
-	// Fatal Errors
-	// See: https://github.com/ethereum/go-ethereum/blob/b9df7ecdc3d3685180ceb29665bab59e9f614da5/core/tx_pool.go#L516
-	gethFatal = regexp.MustCompile(`(: |^)(exceeds block gas limit|invalid sender|negative value|oversized data|gas uint64 overflow|intrinsic gas too low|nonce too high)$`)
+	NonceTooLow                       = "NonceTooLow"
+	ReplacementTransactionUnderpriced = "ReplacementTransactionUnderpriced"
+	LimitReached                      = "LimitReached"
+	TransactionAlreadyInMempool       = "TransactionAlreadyInMempool"
+	TerminallyUnderpriced             = "TerminallyUnderpriced"
+	InsufficientEth                   = "InsufficientEth"
+	TooExpensive                      = "TooExpensive"
+	Fatal                             = "Fatal"
 )
 
-var hexDataRegex = regexp.MustCompile(`0x\w+$`)
+type ClientErrors = map[string]*regexp.Regexp
 
-// IsReplacementUnderpriced indicates that a transaction already exists in the mempool with this nonce but a different gas price or payload
-func (s *SendError) IsReplacementUnderpriced() bool {
+// Parity
+// See: https://github.com/openethereum/openethereum/blob/master/rpc/src/v1/helpers/errors.rs#L420
+var parFatal = regexp.MustCompile(`^Transaction gas is too low. There is not enough gas to cover minimal cost of the transaction|^Transaction cost exceeds current gas limit. Limit:|^Invalid signature|Recipient is banned in local queue.|Supplied gas is beyond limit|Sender is banned in local queue|Code is banned in local queue|Transaction is not permitted|Transaction is too big, see chain specification for the limit|^Invalid RLP data`)
```

Comment:
Needs `$`.

also didn't no that pipe worked without match groups - is this tested?

Reply:
I tried to follow the same format as `gethFatal` but wasn't working. This format actually works